File: members/U02CDR60PFS/programmers/weekly-challenge/모음사전.js
const alpha = ["A", "E", "I", "O", "U"];
const dict = {};
let order = 0;

const dfs = (cnt, word) => {
  if (cnt === 5) return;

  for (let i = 0; i < 5; i++) {
    dict[word + alpha[i]] = ++order;
    dfs(cnt + 1, word + alpha[i]);
  }
};

function solution(word) {
  dfs(0, "");
  return dict[word];
}

File: members/U02B9E86MST/Programmers/Weekly/모음사전.cpp
#include <string>
#include <vector>
#include <set>

using namespace std;
set<string> words;
char order[5] = {'A','E','I','O','U'};
void recursion(string word) {
    if(word.length()==5) {
        words.insert(word);
        return;
    }
    
    for(int i=-1; i<5; i++){
        if(i==-1) {
            words.insert(word);
            continue;
        }
        recursion(word+order[i]);
    }
}

int getIndex(string word) {
    int index = 0;
    for(auto x : words) {
        if(x.compare(word) == 0) return index;
        ++index;
    }
}

int solution(string word) {
    recursion("");
    return getIndex(word);
}
